Utils/TimeConstants.swift

```diff
@@ -32,6 +32,10 @@ extension NSDate {
         return NSDate(timeIntervalSince1970: Double(timestamp) / 1000)
     }
 
+    public class func fromMicrosecondTimestamp(microsecondTimestamp: MicrosecondTimestamp) -> NSDate {
```

Comment:
Take a quick skim through the existing code (look for `1000000`), see if this should be used anywhere else.